### PR TITLE
Health system with accessors

### PR DESCRIPTION
**Pull request by maximatorax**
When player die, a temporary GameOver scene load, which only say "You suck!"
When player take his job, his MaxHealth is increased to 110 instead of 100(temp values), it also heal him to his MaxHealth.
The player can't overheal himself.
The player can only have 1 job, preventing him from abusing the healing from taking a job

### CODE REVIEW

File: src/Assets/Scripts/AttackSystem/AttackSystem.cs [hunk 1]
```diff
@@ -14,13 +14,19 @@ public class AttackSystem : MonoBehaviour,IAttackSystem
 
     public Attack Ability;
 
+    CharacterController charCtrl;
```
Comment (CowMadness) [inline src/Assets/Scripts/AttackSystem/AttackSystem.cs]: Rename to characterControl (way more clear)

File: src/Assets/Scripts/AttackSystem/AttackSystem.cs [hunk 2]
```diff
@@ -76,4 +97,12 @@ void IAttackSystem.ChangeAttack(Attack attackToChange, Attack newAttack)
     {
         ChangeAttack(attackToChange, newAttack);
     }
+
+    void OnDrawGizmosSelected()
```
Comment (CowMadness) [inline src/Assets/Scripts/AttackSystem/AttackSystem.cs]: What is this used for?

Comment (maximatorax) [reply src/Assets/Scripts/AttackSystem/AttackSystem.cs]: It's used to see the Spherecast in editor by using a drawing gizmos, to visualy see the range of the attack in editor

**Comment by maximatorax** [comment]
Fixed according to comment, merging

**Pull request merged**